webapp/handlers.py

```diff
@@ -0,0 +1,93 @@
+import socket
+from datetime import datetime
+from urllib.parse import unquote, urlparse, urlunparse
+
+import flask
+import webapp.template_utils as template_utils
+
+def set_handlers(app):
+    @app.context_processor
+    def utility_processor():
+        """
+        This defines the set of properties and functions that will be added
+        to the default context for processing templates. All these items
+        can be used in all templates
+        """
+
+        return {
+            # Functions
+            "format_date": template_utils.format_date,
```

Comment:
Since they're not used in this app, I would remove `format_date`, `replace_admin` and `truncate_chars`, along with their accompanying functions from `template_utils.py`.

`get_year` and `versioned_static` are presumably only used because you copied the markup for error pages from jp.ubuntu.com, and so when you split out `index.html` markup into includes so it can be shared with error pages, that old markup will go and you'll no longer be using those context processors either.

In which case you could delete this whole context_processor function, and all of `templare_utils.py`.

Reply:
Heh I had originally done this but they were required by the base templates. I'll see what we can do about syncing them with jaas.ai and then see what we can remove. 👍

Reply:
See https://github.com/juju/juju-gui/pull/3908#discussion_r269739820. We should ask a designer, but I don't agree that simply using jaas.ai's layout is a good move.